proposed/2022/pack-referenced-projects.md

```diff
@@ -0,0 +1,71 @@
+# Pack referenced projects
+
+- Author Name [maxkoshevoi](https://github.com/maxkoshevoi)
+- Start Date (2022-03-28)
+- GitHub Issue [3891](https://github.com/NuGet/Home/issues/3891)
+- GitHub PR (-)
+
+## Summary
+
+Allow referenced projects' artifacts to be packed inside the NuGet.
+
+## Motivation
+
+When SDK-style project is being packed as a NuGet, all projects that it references are considered as NuGet themselves that this NuGet depends on.
+This is not always a desired behavior, sometimes user needs artifacts from referenced project(s) to be packed into the NuGet.
+Currently there's no easy way to override this behavior except for ditching `csproj`, and going back to `nuspec`. In many cases that involves piping lots of information from the `MSBuild` context into the `nuspec` evaluation via `key=value` parameters.  
+Chaining together dozens of arguments to achieve parity with metadata usually automagically introduced by the SDK becomes tedious, which eventually weakens the otherwise rich set of information attached to a modern NuGet package.
+
+## Explanation
+
+### Functional explanation
+
+Proposed solution is to add new `Pack` and `PackagePath` properties to `ProjectReference` (`<ProjectReference Include="..." Pack="True" PackagePath="tools" />`) and `PackageReference`. 
+If `Pack` is `False` or not present - consider the project (or package) as a NuGet's dependency, if it's `True` - pack its artifacts in.
+If `PackagePath` is not present - pack artifacts to the default location within the package. If it's present - pack them to the specified path.
+
+For `PackageReference`s it's useful in case referenced package is from private feed, but NuGet that's being packed will be public.
+
+### Technical explanation
+
+When NuGet is being created from `csproj` file:
+
+1) If `ProjectReference` doesn't have `Pack` property, or has it set to `False` - preserve behavior that have existed before introduction of this feature (don't include artifacts of referenced projects into the NuGet, and consider them as NuGet-dependencies).
+2) If `ProjectReference` has `Pack` property and it's set to `True` - include it's artifacts into the NuGet (to the default location or one specified in `PackagePath`).
+3) Do the same steps for `PackageReference`s
+
+## Drawbacks
```

Comment:
It may be worth to add that one of the best practices in NuGet today is to have one assembly per NuGet package. 

https://docs.microsoft.com/en-us/nuget/create-packages/creating-a-package#decide-which-assemblies-to-package

https://docs.microsoft.com/en-us/nuget/create-packages/select-assemblies-referenced-by-projects#recommended-one-assembly-per-package

https://docs.microsoft.com/en-us/nuget/create-packages/select-assemblies-referenced-by-projects#not-recommended-multiple-assemblies-in-one-package

Reply:
I think some of the feedback around compatibiltiy and redistribution of packages should be listed in the drawbacks here.